src/Manipulators/Background.php

```diff
@@ -26,9 +26,10 @@ public function run(Image $image)
         $color = (new Color($this->bg))->formatted();
 
         if ($color) {
-            $new = $image->getDriver()->newImage($image->width(), $image->height(), $color);
-            $new->mime = $image->mime;
-            $image = $new->insert($image, 'top-left', 0, 0);
+            $new = $image->driver()->createImage($image->width(), $image->height())->fill($color);
+            // TODO: Find a way to communicate the mime
```

Comment:
Perhaps Image::encodeByMediaType() is what we need.

Reply:
Or Image::encodeByExtension().

Reply:
@ADmad I was able to fix with the following approach here https://github.com/Art4/glide/pull/1/commits/f51a6278e881db70ec5c1b8a18b0e1283a8f2bba